src/YagrCore/plugins/tooltip/tooltip.ts

```diff
@@ -562,9 +566,60 @@ class YagrTooltip {
         }
     };
 
-    private onMouseUp = () => {
+    /**
+     * Calculates where exactly cursor leaved the chart
+     * and sets range[1] to this position
+     */
+    private setCursorLeaved = (e: MouseEvent) => {
+        const rect = this.over.getBoundingClientRect();
+        const x = e.clientX;
+        const range = this.state.range!;
+        const startPoint = range[0]!;
+        const xInOver = x - rect.left;
+        const end = xInOver > startPoint.clientX;
+        const timeline = this.yagr.config.timeline;
+
+        let result;
+        if (end) {
+            range[1] = {
+                clientX: this.bWidth,
+                value: this.yagr.uplot.posToVal(this.bWidth, 'x'),
+                idx: timeline.length - 1,
+            };
+            result = range[1];
+        } else {
+            /** Swap range[1] and range[0] in case if tooltip leaved chart in begining of element */
+            range[1] = range[0];
+            range[0] = {
+                clientX: 0,
+                value: this.yagr.uplot.posToVal(0, 'x'),
+                idx: 0,
+            };
+
+            result = range[0];
+        }
+
+        this.yagr.uplot.setCursor({
+            left: result.clientX,
+            top: e.clientY - rect.top,
+        });
+
+        return result;
```

Comment:
It seems that we shouldn't return a value, because we don't use it

Reply:
Totally!